### PR TITLE
[RFR] Link function

### PR DESCRIPTION
**Pull request by djhi**
- [x] Refactored transformations to introduce context
- [x] Applied transformations by stages (pre, post)
- [x] Implemented `LINK` function

### CODE REVIEW

File: src/api/controller/api/publish.js [hunk 1]
```diff
@@ -14,15 +13,32 @@ export const publishMiddleware = async (ctx) => {
...
-        const transformedDataset = await Promise.all(dataset.map(transformDocument));
+        const transformedDataset = await Promise.all(dataset.map(transformDocumentPreStage));
         await ctx.publishedDataset.insertMany(transformedDataset);
         handled += dataset.length;
     }
 
+    const transformDocumentPostStage = transformDocumentFactory('post');
+    handled = 0;
+
+    while (handled < count) {
```
Comment (ThieryMichel) [inline src/api/controller/api/publish.js:34]: Should'nt the uri be added at the start, and then th Link operation could be executed at the same time as all the other.

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

File: src/api/controller/api/publish.js (hunk 1 above)
Comment (djhi) [reply src/api/controller/api/publish.js:34]: I'm not sure, can't it be the result of other transformed fields too ?

Comment (djhi) [reply src/api/controller/api/publish.js:34]: Which also means it shouldn't be added automatically as the first field too btw

**Comment by djhi** [comment]
Superceded by #33

**Pull request closed without merging**